reporters/reporter-default/src/main/java/org/radargun/reporting/html/HtmlReporter.java

```diff
@@ -147,47 +172,109 @@ private void writeTestReportDocuments(Set<String> combinedTests, Map<String, Lis
          }
          TestAggregations ta = new TestAggregations(entry.getKey(), entry.getValue());
          TestReportDocument testReport = new TestReportDocument(ta, targetDir, testReportConfig);
+
+         freemarker.template.Configuration cfg = initConfig();
          try {
-            testReport.open();
-            testReport.writeTest();
-         } catch (IOException e) {
-            log.error("Failed to create test report " + entry.getKey(), e);
-         } finally {
-            testReport.close();
+            testReport.createReportDirectory();
+            testReport.createTestCharts();
+
+            Template reportTemplate = cfg.getTemplate("testReport.ftl");
+            PrintWriter printWriter = new PrintWriter(targetDir + File.separator + "test_" + testReport.testName + ".html");
+
+            Map root = new HashMap();
+            root.put("testReport", testReport);
+            root.put("enums", DefaultObjectWrapper.getDefaultInstance().getEnumModels());
+            reportTemplate.process(root, printWriter);
+            printWriter.close();
```

Comment:
Hi Vitalii, no need to repeat almost everything that @mcimbora commented a few days ago. That doesn't help anybody. Please comment on things that are not yet obvious otherwise it just spams everyone's email inbox.